src/model/index.ts

```
@@ -0,0 +1,87 @@
+/**
+ * This file is part of Serlo.org API
+ *
+ * Copyright (c) 2020 Serlo Education e.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @copyright Copyright (c) 2020 Serlo Education e.V.
+ * @license   http://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/serlo-org/api.serlo.org for the canonical source repository
+ */
+import { RequestInit } from 'node-fetch'
+
+import { Instance } from '../types'
+import { Cache } from './cache'
+import { LockManager } from './lock-manager'
+
+type Fetch<Result = unknown> = (
+  init: RequestInit & { path: string }
+) => Promise<Result>
+
+export interface Model {
+  update(key: string): Promise<void>
```

It would probably make sense to also return the (optional) new value or error. Then the consumer can decide what to do with it (retry our ignore exceptions).